lib/src/events/thread_create_event.dart

```diff
@@ -22,3 +22,24 @@ class ThreadCreateEvent implements IThreadCreateEvent {
     }
   }
 }
+
+abstract class IThreadUpdateEvent {
+  /// The thread that was just created
+  IThreadChannel get thread;
+}
+
+/// Fired when a thread is created
+class ThreadUpdateEvent implements IThreadUpdateEvent {
+  /// The thread that was just created
+  @override
+  late final IThreadChannel thread;
+
+  /// Creates an instance of [ThreadUpdateEvent]
+  ThreadUpdateEvent(RawApiMap raw, INyxx client) {
+    thread = ThreadChannel(client, raw["d"] as RawApiMap);
+
+    if (client.cacheOptions.channelCachePolicyLocation.event && client.cacheOptions.channelCachePolicy.canCache(thread)) {
+      client.channels[thread.id] = thread;
+    }
+  }
+}
```

Comment:
I don't know if you already planned this, but that wouldn't be mean to add an `oldThread` property by checking the cache.

The same could apply to `IThreadMemberUpdateEvent`

Reply:
Good point - it passed through my mind while writing it but I completely forgot! Guess I was a bit too tired :P

I'll add these tomorrow when I have time.